src/js/actions/documents.js

```diff
@@ -856,12 +860,119 @@ define(function (require, exports) {
     beforeStartup.writes = [locks.JS_DOC];
 
     /**
-     * Initialize the inactive documents. (The active document is initialized beforeStartup.)
+     * Send info to search store about searching for documents and
+     * initialize the inactive documents. (The active document is initialized beforeStartup.)
      *
      * @param {{currentIndex: number, docCount: number}=} payload
      * @return {Promise}
      */
     var afterStartup = function (payload) {
+        var appStore = this.flux.store("application"),
+            docStore = this.flux.store("document");
+
+        /**
+         * Make list of currently open documents to be used as search options
+         * 
+         * @return {Immutable.List.<object>}
+        */
+        var currDocSearchOptions = function () {
+            var document = appStore.getCurrentDocument(),
+                openDocs = appStore.getOpenDocumentIDs().filterNot(function (doc) {
+                                return doc === document.id;
+                            }),
+                docMap = openDocs.map(function (doc) {
+                    return {
+                        id: "curr-doc_" + doc.toString(),
+                        title: docStore.getDocument(doc).name,
+                        type: "item",
+                        category: [strings.SEARCH.CATEGORIES.DOCUMENT, strings.SEARCH.CATEGORIES.CURRENT]
+                    };
+                }),
+                docLabel = {
+                    id: "curr-doc_header",
+                    title: strings.SEARCH.HEADERS.CURRENT_DOC,
+                    type: "header"
+                },
+
+                docOptions = docMap.unshift(docLabel);
```

Comment:
Also, what happens if one day we decide we don't want headers. If your function took care of this in one central location, we would have to change it in only one place, and clients of the Search API you're developing wouldn't need to worry about parity between different modules.